# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderBridgeChild.cpp
# Commit: 898ed02804fe
# Full Hash: 898ed02804fefc9786112f135b9621cc444e6860
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-05-23 15:52:15
# Regressor Bug: 1550422
# File Overlap Count: 1
# Description:
#   Bug 1550422 - P15. Move Skip and Once gfxPrefs to StaticPrefs. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31259
# ==============================================================================

diff -r e1b7abc99ae9 -r 898ed02804fe gfx/layers/wr/WebRenderBridgeChild.cpp
--- a/gfx/layers/wr/WebRenderBridgeChild.cpp	Wed May 22 12:44:04 2019 +0000
+++ b/gfx/layers/wr/WebRenderBridgeChild.cpp	Wed May 22 22:27:37 2019 +0000
@@ -7,7 +7,7 @@
 #include "mozilla/layers/WebRenderBridgeChild.h"
 
 #include "gfxPlatform.h"
-#include "gfxPrefs.h"
+#include "mozilla/StaticPrefs.h"
 #include "mozilla/dom/TabGroup.h"
 #include "mozilla/layers/CompositableClient.h"
 #include "mozilla/layers/CompositorBridgeChild.h"
@@ -73,8 +73,9 @@
 
 void WebRenderBridgeChild::AddWebRenderParentCommand(
     const WebRenderParentCommand& aCmd, wr::RenderRoot aRenderRoot) {
-  MOZ_ASSERT(aRenderRoot == wr::RenderRoot::Default ||
-             (XRE_IsParentProcess() && gfxPrefs::WebRenderSplitRenderRoots()));
+  MOZ_ASSERT(
+      aRenderRoot == wr::RenderRoot::Default ||
+      (XRE_IsParentProcess() && StaticPrefs::WebRenderSplitRenderRoots()));
   mParentCommands[aRenderRoot].AppendElement(aCmd);
 }
 
@@ -119,7 +120,7 @@
   for (auto& renderRoot : aRenderRoots) {
     MOZ_ASSERT(
         renderRoot.mRenderRoot == wr::RenderRoot::Default ||
-        (XRE_IsParentProcess() && gfxPrefs::WebRenderSplitRenderRoots()));
+        (XRE_IsParentProcess() && StaticPrefs::WebRenderSplitRenderRoots()));
     renderRoot.mCommands = std::move(mParentCommands[renderRoot.mRenderRoot]);
   }
 
@@ -151,7 +152,7 @@
   for (auto& update : aRenderRootUpdates) {
     MOZ_ASSERT(
         update.mRenderRoot == wr::RenderRoot::Default ||
-        (XRE_IsParentProcess() && gfxPrefs::WebRenderSplitRenderRoots()));
+        (XRE_IsParentProcess() && StaticPrefs::WebRenderSplitRenderRoots()));
     update.mCommands = std::move(mParentCommands[update.mRenderRoot]);
   }
 
@@ -175,7 +176,7 @@
   for (auto renderRoot : wr::kRenderRoots) {
     if (!mParentCommands[renderRoot].IsEmpty()) {
       MOZ_ASSERT(renderRoot == wr::RenderRoot::Default ||
-                 gfxPrefs::WebRenderSplitRenderRoots());
+                 StaticPrefs::WebRenderSplitRenderRoots());
       this->SendParentCommands(mParentCommands[renderRoot], renderRoot);
       mParentCommands[renderRoot].Clear();
     }